src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -206,7 +207,42 @@ internal static class MethodGenerator
         {
             return !method.ExplicitInterfaceImplementations.Any() && !method.IsOverride
                 ? method.TypeParameters.GenerateConstraintClauses()
-                : default;
+                : GenerateDefaultConstraints(method);
+        }
+
+        private static SyntaxList<TypeParameterConstraintClauseSyntax> GenerateDefaultConstraints(IMethodSymbol method)
+        {
+            Debug.Assert(method.ExplicitInterfaceImplementations.Any() || method.IsOverride);
+            var seenTypeParameters = new HashSet<string>();
+            var listOfClauses = new List<TypeParameterConstraintClauseSyntax>(method.TypeParameters.Length);
+            foreach (var parameter in method.Parameters)
+            {
+                if (parameter.Type is not ITypeParameterSymbol { NullableAnnotation: NullableAnnotation.Annotated } typeParameter ||
+                    !seenTypeParameters.Add(parameter.Type.Name))
```

Comment:
oh please don't mix a check and a mutation in the same expression.  separate out to two if blocks.

Reply:
@CyrusNajmabadi Will do. But curious if there is any particular reason?

Reply:
99.999% of all checks are non-mutating (true number :)).  As such, it can be easy to miss and get confused by the odd case which looks like a normal if check but does mutate.  I like uncommon cases to stick out so it is very easy to see and understand what is going on.